extension/wallet/src/protocol/make_create_swap_payload.rs

```diff
@@ -0,0 +1,158 @@
+}
+
+/// Constructs a new [`CreateSwapPayload`] with the given Bitcoin amount.
+///
+/// This will select UTXOs from the wallet to cover the given amount.
+#[wasm_bindgen]
+pub async fn make_sell_create_swap_payload(
+    wallet_name: String,
+    btc: String,
+) -> Result<JsValue, JsValue> {
+    let btc = map_err_from_anyhow!(Amount::from_str_in(&btc, Denomination::Bitcoin))?;
+
+    let payload = map_err_from_anyhow!(make_sell_create_swap_payload_w(wallet_name, btc).await)?;
+    let payload = map_err_from_anyhow!(JsValue::from_serde(&payload))?;
+
+    Ok(payload)
+}
+
+async fn make_sell_create_swap_payload_w(
+    wallet_name: String,
+    sell_amount: Amount,
+) -> Result<CreateSwapPayload, Error> {
+    let btc_asset_id = {
+        let guard = BTC_ASSET_ID.lock().expect_throw("can get lock");
+        *guard
+    };
+
+    make_create_swap_payload(wallet_name, sell_amount, btc_asset_id, btc_asset_id).await
+}
+
+async fn make_create_swap_payload(
+    wallet_name: String,
+    sell_amount: Amount,
+    sell_asset: AssetId,
+    fee_asset: AssetId,
+) -> Result<CreateSwapPayload, Error> {
+    let (bobs_fee_rate, fee_offset) = if fee_asset == sell_asset {
+        // Bob currently hardcodes a fee-rate of 1 sat / vbyte, hence
+        // there is no need for us to perform fee estimation. Later
+        // on, both parties should probably agree on a block-target
+        // and use the same estimation service.
+        let bobs_fee_rate = Amount::ONE_SAT;
+        let fee_offset = calculate_fee_offset(bobs_fee_rate);
+
+        (bobs_fee_rate.as_sat() as f32, fee_offset)
+    } else {
+        (0.0, Amount::ZERO)
+    };
+
+    let client = {
+        let guard = ESPLORA_API_URL.lock().expect_throw("can get lock");
+        EsploraClient::new(guard.clone())
+    };
+
+    let (outputs, secret_key, address, blinding_key) = {
+        let mut wallet = current(&wallet_name, &LOADED_WALLET)
+            .await
+            .map_err(Error::LoadWallet)?;
+        wallet.sync(&client).await.map_err(Error::SyncWallet)?;
+        let outputs = wallet
+            .coin_selection(sell_amount, sell_asset, bobs_fee_rate, fee_offset)
+            .map_err(Error::CoinSelection)?;
+        (
+            outputs,
+            wallet.secret_key(),
+            wallet.address(),
+            wallet.blinding_secret_key(),
+        )
+    };
+
+    Ok(CreateSwapPayload {
+        address,
+        alice_inputs: outputs
+            .into_iter()
+            .map(|utxo| SwapUtxo {
+                outpoint: utxo.txin,
+                blinding_key,
+            })
+            .collect(),
+        amount: sell_amount,
+    })
+}
+
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    #[error("Wallet is not loaded: {0}")]
+    LoadWallet(anyhow::Error),
+    #[error("Could not sync waller: {0}")]
+    SyncWallet(anyhow::Error),
+    #[error("Coin selection: {0}")]
+    CoinSelection(anyhow::Error),
+}
```

Comment:
Worth looking into the configuration options of `thiserror`: https://docs.rs/thiserror/1.0.26/thiserror/index.html#details

Instead of printing the error inline using `{0}`, you would be better of annotating it with `#[source]` to print it as part of the cause chain.

Reply:
leaving this to the platform team